src/compiler/checker.ts

```diff
@@ -26481,13 +26466,12 @@ namespace ts {
                 // already assigned contextual types.
                 if (!(links.flags & NodeCheckFlags.ContextChecked)) {
                     links.flags |= NodeCheckFlags.ContextChecked;
-                    if (contextualSignature) {
-                        const type = getTypeOfSymbol(getMergedSymbol(node.symbol));
-                        if (isTypeAny(type)) {
-                            return;
-                        }
-                        const signature = getSignaturesOfType(type, SignatureKind.Call)[0];
-                        if (isContextSensitive(node)) {
+                    const signature = firstOrUndefined(getSignaturesOfType(getTypeOfSymbol(getSymbolOfNode(node)), SignatureKind.Call));

```

Comment:
The internal singleton `anyFunctionType` exists, is function-y, and lacks call signatures, no? Our checker internal `isFunctionType` helper also fails to account for this; but it definitely seems correct to do so. Maybe amending `isFunctionType` and using that here would be a good idea, rather than duplicating the check in a slightly different manner?

Reply:
The `anyFunctionType` will never be captured as the type of a symbol, so we'll never see it here.

Reply:
`checkObjectLiteralMethod` can return `anyFunctionType` when `CheckMode.SkipGenericFunctions` is set, which is called by `checkObjectLiteral` and then cached on the transient member symbol we produce directly as the type.

So yea, we can totally see it as the type of a symbol.

Reply:
I'm not sure what we're debating here. I think the code to extract the signature is fine as is, I'm not getting what change you want and why?

Reply:
From our discussion in person: Because the symbol is a binder symbol (or merge/late bound symbol) of a method, it'll never have a type affixed directly to it by `checkObjectLiteral` (it'll clone the symbol first); so the only way it could happen would be if the `anyFunctionType` leaked out of an inference phase via another cache, which we'd consider a bug regardless, so it _probably_ doesn't need to be accounted for here, assuming the invariants around symbols which we expect to be upheld are.